content/blog/2025/fluid-type.md

```diff
@@ -0,0 +1,356 @@
+  it has been 'best practice'
+  to use relative units
+  (especially `em` and `rem`)
+  for sizing text.
+  That's great!
+  But after playing around with my user preferences,
+  I think we can improve on the common approaches.
+---
+
+{% import "quotes.macros.njk" as quotes %}
+
+{% callout 'note' %}
+We talked about some of these ideas
+in our [Winging It](/2025/01/24/winging-it-15/) live stream
+last month,
+with Alan Stearns --
+co-chair of the CSS Working Group.
+Check that out for more discussion
+of typography and CSS units!
+{% endcallout %}
+
+The browser provides a default text size
+based on user preferences,
+and our text should be _relative to that preference_.
+Establishing our root font-size with an `em` value
+helps keep that relationship intact.
+
+Similarly,
+font 'scales' are intended to
+maintain some _internal harmony_
+across a range of different font sizes.
+By establishing our entire scale in relative units
+like `em` and `rem`,
+we maintain a consistent relationship between sizes,
+while allowing the entire system
+to scale up or down
+relative to user defaults
+or browser [(page) zoom](/2024/07/09/zoomies/)
+
+Recently,
+'fluid typography' has pushed us
+to incorporate viewport or container relative units
+into our font sizing as well --
+generally with a `clamp()` function
+to integrate both `em`/`rem` and `vi`/`cqi` _inputs_
+in a responsive algorithm
+with lower and upper boundaries.
+See [utopia.fyi](https://utopia.fyi)
+for the most common approach:
+
+```css
+:root {
+  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem)
+}
+```
+
+This comes with some accessibility risks
+if we're not careful,
+so keep an eye on the warnings
+that Utopia provides.
+Or jump over to [fluid.style](https://fluid.style)
+to test the page-zoom capabilities
+of any fluid calculation.
+
+This utopian approach works pretty well,
+but the more I play with it
+the more questions I have.
+So let's dig in,
+and see if there are improvements we can make.
+
+
+## What's the purpose of a font-size preference?
+
+I've been
+[playing with my browser settings](https://www.miriamsuzanne.com/2024/01/24/have-preferences/),
+starting with fonts
+and then the default font-size.
+This is a setting that most browsers provide.
+Since I generally prefer web text around `24px`,
+that seems like an obvious default, right?
+
+But I've already designed my site
+to have large text _relative to_ the user default.
+Ignoring the fluid bits for now,
+I set the root `font-size` to roughly `1.5em` --
+half again the size of the user setting.
+When my preference was set to the default `16px`,
+my site would already display text at `24px`.
+Now that I've said I _want_ `24px` font-size,
+my site has increased to `36px`!
+
+It seems like Chromium browsers
+no longer allow setting a specific font size.
+Instead you can choose from the default `medium`
+or a list of larger and smaller options:
+
+- Very small
+- Small
+- Medium (recommended)
+- Large
+- Very large
```

Comment:
It does show the size when you go into the Customize Fonts panel and focus or hover over the Font Size slider
![font-size-slider](https://github.com/user-attachments/assets/6af4194b-0fc6-470b-9629-71df311099e2)

Reply:
Ah, that's interesting. I guess I hadn't gone deeper.